From 84555eb9d139dcdc7c3c80a0fc36230c18e71025 Mon Sep 17 00:00:00 2001
From: Weixin Zhou <zwx@rock-chips.com>
Date: Mon, 23 Nov 2020 20:03:31 +0800
Subject: [PATCH] arm64: dts: rockchip: rk3566-rk817-tablet-k108: adjust sdcard
 config

if sdcard detect pin active status is low, forcejtag must be 0.

Signed-off-by: Weixin Zhou <zwx@rock-chips.com>
Change-Id: I653693e92422b191539d11ba498a4c14a4ccd117
---
 .../arm64/boot/dts/rockchip/rk3566-rk817-tablet-k108.dts | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3566-rk817-tablet-k108.dts b/arch/arm64/boot/dts/rockchip/rk3566-rk817-tablet-k108.dts
index 852bbc41a891..23a6e0188d30 100644
--- a/arch/arm64/boot/dts/rockchip/rk3566-rk817-tablet-k108.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3566-rk817-tablet-k108.dts
@@ -1067,6 +1067,12 @@
 		};
 	};
 
+	sdmmc0 {
+		sdmmc0_det_gpio: sdmmc0-det-gpio {
+			rockchip,pins = <0 RK_PA4 RK_FUNC_GPIO &pcfg_pull_up>;
+		};
+	};
+
 	usb {
 		vcc5v0_host_en: vcc5v0-host-en {
 			rockchip,pins = <3 RK_PA3 RK_FUNC_GPIO &pcfg_pull_none>;
@@ -1168,8 +1174,9 @@
 	sd-uhs-sdr104;
 	vmmc-supply = <&vcc_sd>;
 	vqmmc-supply = <&vccio_sd>;
+	cd-gpios = <&gpio0 RK_PA4 GPIO_ACTIVE_HIGH>;
 	pinctrl-names = "default";
-	pinctrl-0 = <&sdmmc0_bus4 &sdmmc0_clk &sdmmc0_cmd &sdmmc0_det>;
+	pinctrl-0 = <&sdmmc0_bus4 &sdmmc0_clk &sdmmc0_cmd &sdmmc0_det_gpio>;
 	status = "okay";
 };
 
-- 
2.35.3

